nodejs/week3/homework/homeworkWeek3.js

```diff
@@ -0,0 +1,54 @@
+import knex from "knex";
+const knexInstance = knex({
+  client: "mysql2",
+  connection: {
+    host: process.env.DB_HOST || "127.0.0.1",
+    port: process.env.DB_PORT || 3306,
+    user: process.env.DB_USER || "root",
+    password: process.env.DB_PASSWORD || "pass",
+    database: process.env.DB_NAME || "hyf_node_week3_warmup",
+    multipleStatements: true,
+  },
+});
+
+import express from "express";
+const app = express();
+const port = process.env.PORT || 3000;
+
+app.use(express.json());
+
+const apiRouter = express.Router();
+app.use("/api", apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use("/contacts", contactsAPIRouter);
+
+//VULNERABLE TO INJECTION ATTACKS
+contactsAPIRouter.get("/", async (req, res) => {
+  let query = knexInstance.select("*").from("contacts");
+
+  const sanitizedOrderBy = knexInstance.raw("??", [orderBy]); // Escaping identifiers query = query.orderByRaw(sanitizedOrderBy);
```

Comment:
I'm unsure if I understand this line, can you explain please?

Reply:
The correct position was further down. This is the fourth way I've found to sanitize the code,and my favorite. Knex treats user input as an identifier, so no matter what you write or try to comment out, it will always be handled safely as a string. 🙂:)
select * from `contacts` order by `first_name;DROP TABLE contacts`